datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -224,6 +226,62 @@ impl<S: SimplifyInfo> ExprSimplifier<S> {
     }
 }
 
+#[allow(rustdoc::private_intra_doc_links)]
+/// Canonicalize any BinaryExprs that are not in canonical form
+/// <literal> <op> <col> is rewritten to <col> <op> <literal> (remember to switch the operator)
+/// <col> <op> <literal> is canonical
+/// <col1> <op> <col2> is rewritten so that the name of col1 sorts higher than col2 (b > a would be canonicalized to a < b)
```

Comment:
Might need to upgrade the docs here, for example remove the `remember to switch the operator` line (or change it to clarify that operator would be switched)